compiler_gym/envs/mlir/datasets/matmul.py

```diff
@@ -0,0 +1,294 @@
+        return benchmark
+
+    @memoized_property
+    def sources(self) -> Iterable[BenchmarkSource]:
+        return [
+            BenchmarkSource(filename="source.mlir", contents=self._src),
+        ]
+
+    @property
+    def source(self) -> str:
+        """Return the single source file contents as a string."""
+        return self._src.decode("utf-8")
+
+
+class MatmulDataset(Dataset):
+    """A dataset which generates matmul programs."""
+
+    def __init__(
+        self,
+        site_data_base: Path,
+        sort_order: int = 0,
+        matmul_bin: Optional[Path] = None,
+        matmul_includes: Optional[Path] = None,
+    ):
+        """Constructor.
+        :param site_data_base: The base path of a directory that will be used to
+            store installed files.
+        :param sort_order: An optional numeric value that should be used to
+            order this dataset relative to others. Lowest value sorts first.
+        :param matmul_bin: The path of the matmul binary to use. If not
+            provided, the version of matmul shipped with CompilerGym is used.
+        :param matmul_includes: The path of the matmul includes directory. If
+            not provided, the includes of the matmul shipped with CompilerGym is
+            used.
+        """
+        super().__init__(
+            name="generator://matmul-v0",
+            description="Targeted size matmul programs",
+            references={},
+            license="donotsubmit",
+            site_data_base=site_data_base,
+            sort_order=sort_order,
+            benchmark_class=MatmulBenchmark,
+        )
+        self.matmul_bin_path = matmul_bin or _matmul_BIN
+        self.matmul_includes_path = matmul_includes or _matmul_INCLUDES
+        """# The command that is used to compile an mlir-IR bitcode file from a
+        # matmul input. Reads from stdin, writes to stdout.
+        self.clang_compile_command: List[str] = ClangInvocation.from_c_file(
+            "-",  # Read from stdin.
+            copt=[
+                "-xc",  # The C programming language.
+                "-ferror-limit=1",  # Stop on first error.
+                "-w",  # No warnings.
+                f"-I{self.matmul_includes_path}",  # Include the matmul headers.
+            ],
+        ).command(
+            outpath="-"  # Write to stdout.
+        )"""
+
+    @property
+    def size(self) -> int:
+        # Actually 2^32 - 1, but practically infinite for all intents and
+        # purposes.
+        return len(matmul_sizes)
+
+    def name_from_size(self, mnk):
+        return f"{self.name}/{mnk[0]}_{mnk[1]}_{mnk[2]}"
+
+    # TODO(kyleherndon): Benchmarks are actually dynamically generated for any
+    # provided parameters, figure out a better way to represent this in the list of
+    # available benchmarks
+    def benchmark_uris(self) -> Iterable[str]:
+        return (self.name_from_size(mnk) for mnk in matmul_sizes)
+
+    def benchmark(self, uri: str) -> MatmulBenchmark:
```

Comment:
I believe you could use [benchmark_from_parsed_uri()](https://compilergym.com/compiler_gym/datasets.html#compiler_gym.datasets.Dataset.benchmark_from_parsed_uri) instead of `benchmark()` to save you the string parsing here

Reply:
This part is going to be changed in another PR.